From 1ccf929fc6ac2821a6a55660cf6eaa7fe37de68d Mon Sep 17 00:00:00 2001
From: "Lopez Esparza, Ricardo" <ricardo.lopez.esparza@intel.com>
Date: Tue, 19 Apr 2016 19:18:46 -0500
Subject: [PATCH 09/41] ov8858: W/A to sync exposure settings.

In CHT_MRD there is a sync problem between the new exposure,
and the statistics being reported to AE. This causes the
algoritm to return wrong exposure for the next frames. The delay
allows the old statistics to be correctly reported before
applying a new exposure

Change-Id: I0d3dc96d9fbe9935d8dd989abe94ece075ffa1a2
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-31785
Signed-off-by: Lopez Esparza, Ricardo <ricardo.lopez.esparza@intel.com>
Reviewed-on: https://android.intel.com:443/496800
Reviewed-on: https://android.intel.com:443/504753
---
 drivers/external_drivers/camera/drivers/media/i2c/ov8858.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
index 81c9c2d..71d9e79 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
@@ -368,6 +368,11 @@ static int __ov8858_set_exposure(struct v4l2_subdev *sd, int exposure, int gain,
 	int exp_val, ret;
 	dev_dbg(&client->dev, "%s, exposure = %d, gain=%d, dig_gain=%d\n",
 		__func__, exposure, gain, dig_gain);
+	/* W/A: In CHT_MRD there is a sync problem between the new exposure,
+	 * and the statistics being reported to AE. This delay allows the old
+	 * statistics to be correctly reported before applying a new exposure */
+	if (strcmp(dmi_get_system_info(DMI_BOARD_NAME), CHT_HR_DEV_NAME) != 0)
+		msleep(5);
 
 	/* dig_gain is not applied to sensor. OTP WB data is used instead */
 
-- 
1.9.1

